src/ResourceManager/OperationalInsights/Commands.OperationalInsights/DataSources/NewDataSourceCmdletsPerKind/NewAzureOperationalInsightsAzureAuditDataSourceCommand.cs

```diff
@@ -24,19 +24,37 @@ namespace Microsoft.Azure.Commands.OperationalInsights
         DefaultParameterSetName = ByWorkspaceName), OutputType(typeof(PSDataSource))]
     public class NewAzureOperationalInsightsAzureAuditDataSourceCommand : NewAzureOperationalInsightsDataSourceBaseCmdlet
     {
+        [Parameter(Position = 0, ParameterSetName = ByWorkspaceObject, Mandatory = true, ValueFromPipeline = true,
+            HelpMessage = "The workspace that will contain the data source.")]
+        [ValidateNotNull]
+        public override PSWorkspace Workspace { get; set; }
+
+        [Parameter(Position = 1, ParameterSetName = ByWorkspaceName, Mandatory = true, ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource group name.")]
+        [ValidateNotNullOrEmpty]
+        public override string ResourceGroupName { get; set; }
+
+        [Parameter(Position = 2, ParameterSetName = ByWorkspaceName, Mandatory = true, ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The name of the workspace that will contain the data source.")]
+        [ValidateNotNullOrEmpty]
+        public override string WorkspaceName { get; set; }
+
+        [Parameter(Position = 3, Mandatory = true, ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The data source name.")]
+        [ValidateNotNullOrEmpty]
+        public override string Name { get; set; }
+
         [Parameter(Position = 4, Mandatory = true, ValueFromPipelineByPropertyName = true,
         HelpMessage = "The azure subscription Id.")]
         [ValidateNotNullOrEmpty]
         public string SubscriptionId { get; set; }
 
+        [Parameter(Mandatory = false, HelpMessage = "Don't ask for confirmation.")]
+        public override SwitchParameter Force { get; set; }
+
         public override void ExecuteCmdlet()
         {
-            var auditLogProperties = new PSAzureAuditLogDataSourceProperties
-            {
-                SubscriptionId = this.SubscriptionId
-            };
-
-            CreatePSDataSourceWithProperties(auditLogProperties);
+            throw new NotSupportedException("This data source is obsoleted, Please use New-AzureRmOperationalInsightsAzureActivityLogDataSource instead.");
```

Comment:
@haitch this is a breaking change. Existing scripts using this cmdlet will no longer work. Since you are redirecting users to another cmdlet, why not use an alias?